hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/GeneralStage.java

```diff
@@ -162,6 +164,45 @@
             @Nonnull DistributedBiFunction<? super C, ? super T, ? extends Traverser<? extends R>> flatMapFn
     );
 
+    /**
+     * Attaches a rolling aggregation stage. Every input item will be
+     * accumulated using the given {@code aggrOp} and a finished result will be
+     * emitted.
+     * <p>
+     * For example, if the input is {@code {2, 7, 8, -5}} and the aggregate
+     * operation is <em>summing</em>, the output will be {@code {2, 9, 17,
+     * 12}}. The number of input and output items is equal.
+     * <p>
+     * The accumulator is saved to state snapshot; after a restart, the
+     * computation is resumed.
+     * <p>
+     * <strong>NOTE:</strong> Take caution when using ever-growing accumulators
+     * in streaming jobs such as {@code toList()}: they will grow for the
+     * lifetime of the job, even after restart.
+     *
+     * <h3>Limitation on aggregate operations</h3>
+     * The used aggregate operation must not return the accumulator in its
+     * {@linkplain AggregateOperation#finishFn() finish function}. An aggregate
+     * operation is normally allowed to do so, but unlike other stages, this
+     * stage continues to accumulate to the accumulator after finishing
+     * function was called. This stage throws an exception if the finish
+     * function returns the same instance for any accumulator at runtime.
+     * <p>
+     * For example, {@link
+     * com.hazelcast.jet.aggregate.AggregateOperations#summingLong
+     * summingLong()} is OK because its result is an immutable {@code Long}.
+     * {@link com.hazelcast.jet.aggregate.AggregateOperations#toSet() toSet()}
+     * is not because its result is the same {@code Set} instance to which it
+     * accumulates.
```

Comment:
We've got a pretty lame story here. We're basically violating the contract of `AggregateOperation` as it has been defined so far.

Reply:
Yes, we are. But what else? Shall we serde the result? Shall we require the finishFn to return an object that will not refer to the accumulator so that the accumulator can be further accumulated to? I think neither of this... Both will harm performance for an edge case. We add a requirement and check it (partly, only for identity).